src/main/java/seedu/address/storage/RoomOccupancyStorage.java

```diff
@@ -45,18 +54,57 @@ public RoomBook readOnlyRoomOccupancy() throws IOException {
                 rooms.add(room);
                 roomsInArray[i] = room;
             }
-            while (roomsOccupiedFile.hasNextLine()) {
-                int roomNumber = roomsOccupiedFile.nextInt(); //room number that is currently occupied
+            for (int i = 0; i < numberOfRoomsOccupied; i++) {
+                String roomNum = roomsOccupiedFile.next();
+                int roomNumber = Integer.parseInt(roomNum); //room number that is currently occupied
                 Room room = roomsInArray[roomNumber - 1];
                 rooms.remove(room);
                 room.setOccupied(true);
                 roomsInArray[roomNumber - 1] = room;
                 rooms.add(room);
-            }
+            } ;
         } else {
             roomsInArray = null;
             numOfRooms = 0;
         }
-        return new RoomBook(rooms, roomsInArray, numOfRooms, fileNumOfRooms, roomsOccupied);
+        return new RoomBook(rooms, roomsInArray, numOfRooms);
+    }
+
+    /**
+     * Function saves the number of rooms in file according to the input given by user
+     *
+     * @param roomBook contains information of number of rooms as input by user
+     * @param fileNumOfRooms Path to where to write the number of rooms in
+     */
+    public void saveNumberOfRooms(RoomBook roomBook, Path fileNumOfRooms) throws IOException {
+        int numOfRoom = roomBook.getNumOfRooms();
+        FileUtil.createIfMissing(fileNumOfRooms);
+        File file = fileNumOfRooms.toFile();
+        FileWriter fileWriter = new FileWriter(file);
+        fileWriter.write(Integer.toString(numOfRoom));
+        fileWriter.close();
+    }
+
+    /**
+     * Function saves the room numbers of occupied rooms
+     *
+     * @param roomBook contains information of which rooms are occupied
+     * @param fileRoomsOccupied Path to where to write the room numbers of occupied rooms
+     */
+    public void saveOccupiedRooms(RoomBook roomBook, Path fileRoomsOccupied) throws IOException {
+        FileUtil.createIfMissing(fileRoomsOccupied);
+        File file = fileRoomsOccupied.toFile();
+        FileWriter fileWriter = new FileWriter(file);
+        Room[] rooms = roomBook.getRoomsInArray();
+        System.out.println(rooms.length);
+        numberOfRoomsOccupied = 0;
+        for (int i = 0; i < rooms.length; i++) {
+            Room room = rooms[i];
+            if (room.isOccupied()) {
+                numberOfRoomsOccupied++;
+                fileWriter.write(Integer.toString(i + 1) + "\n");
```

Comment:
since you are only counting the number of occupied rooms directly, will there be a problem with this? eg.
i have 3 rooms
* room 1
* room 2
* room 3

So lets say ``room 1`` and ``room 3`` are occupied. so ``numberOfRoomsOccupied = 2`` correct? But next time if you load the ``RoomBook``, it will give me ``room 1`` and ``room 2`` occupied since the ``numberOfRoomsOccupied`` is 2?

Reply:
I dont get your problem here.

Reply:
It wont because 1 and 3 are stored in the hard disk